starlette/testclient.py

```diff
@@ -385,10 +386,15 @@ def __init__(
             raise_server_exceptions=raise_server_exceptions,
             root_path=root_path,
         )
+        if headers is None:
+            headers = {}
+        if "user-agent" not in headers:
```

Comment:
This might be a good place to use `headers.setdefault("user-agent", "testclient")`.